src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSource.cs

```diff
@@ -171,17 +161,26 @@ public HttpSource(PackageSource source, Func<Task<HttpHandlerResource>> messageH
 
         public Task<HttpResponseMessage> GetAsync(Uri uri, ILogger log, CancellationToken token)
         {
-            log.LogInformation(string.Format(CultureInfo.InvariantCulture, _requestLogFormat, "GET", uri));
-
             Func<HttpRequestMessage> requestFactory = () => new HttpRequestMessage(HttpMethod.Get, uri);
 
-            return SendAsync(requestFactory, token);
+            return SendAsync(requestFactory, log, token);
         }
 
-        public async Task<Stream> GetStreamAsync(Uri uri, ILogger log, CancellationToken token)
+        public async Task<Stream> GetStreamAsync(Uri uri, bool ignoreNotFounds, ILogger log, CancellationToken token)
         {
             var response = await GetAsync(uri, log, token);
-            response.EnsureSuccessStatusCode();
+
+            if (ignoreNotFounds && response.StatusCode == HttpStatusCode.NotFound)
+            {
+                response.Dispose();
+                return null;
+            }
+
+            if (!response.IsSuccessStatusCode)
+            {
+                response.Dispose();
+                response.EnsureSuccessStatusCode();
+            }
 
             return await response.Content.ReadAsStreamAsync();
```

Comment:
this code should dispose the response in a

using (var response = await GetAsync(uri, log, token)
{
     if (ignoreNotFounds && response.StatusCode == HttpStatusCode.NotFound)
     {
         return null;
     }

     response.EnsureSuccesStatusCode();
}